src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/CodeActions/CodeActionEndpoint.cs

```diff
@@ -67,6 +71,8 @@ internal sealed class CodeActionEndpoint : IRazorRequestHandler<VSCodeActionPara
             throw new ArgumentNullException(nameof(request));
         }
 
+        var correlationId = Guid.NewGuid();
+        using var __ = _telemetryReporter?.TrackLspRequest(Methods.TextDocumentCodeActionName, LanguageServerConstants.RazorLanguageServerName, correlationId);
```

Comment:
Better to move below the documentContext check so we're not sending telemetry for cases we don't do anything

Reply:
+1. I'd go one further and move it to just above the call to `GetRazorCodeActionsAsync`.